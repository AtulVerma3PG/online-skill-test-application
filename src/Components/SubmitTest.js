import React from "react";
import { logout } from "./utils";
/**
 *Show Candidate Score with details
 *
 * @param {object} props candidate Test details
 */
const SubmitTest = ({ location }) => {
  const { state } = location;
  console.log(state);
  logout();
  return (
    <div>
      <h4>Thanks for you Response.</h4>
      <p>
        Name : {state.firstName} {state.lastName}
        <br />
        Email : {state.email}
        <br />
        Questions Attempted : {state.questionsAttempted} /{" "}
        {state.questionCount}
        <br />
        Score : {state.candidateScore} / {state.questionCount}
      </p>
    </div>
  );
};

SubmitTest.propTypes = {
  location: Object.isRequired,
};

export default SubmitTest;
