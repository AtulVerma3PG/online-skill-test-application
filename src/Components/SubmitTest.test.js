/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import SubmitTest from "./SubmitTest";

const location = {
  hash: "",
    key: "7kdb4z",
  pathname: "/SubmitTest",
  search: "",
  state : {
    questionIndex: 0,
    isFalseLogin: false,
    quizTime: 600,
    questionCount: 0,
    IsQuizSubmitted: false,
    skillLevel: "",
    questionData: "",
    candidateDetails: [],
    candidateResponses: [],
    currentResponse: [],
    questionsAttempted: 0,
    candidateScore: 0,
    firstName: "",
    lastName: "",
    email: "",
    gender: "",
    password: "",
  },
}
test("First Submit Test Snapshot test", () => {
  const component = renderer.create(
    <SubmitTest location= { location }/>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
