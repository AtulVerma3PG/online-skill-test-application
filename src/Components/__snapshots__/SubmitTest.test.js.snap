// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Submit Test Snapshot test 1`] = `
<div>
  <nav
    className="menu-container-nav"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
  </nav>
  <div
    className="thankyou-page"
  >
    <img
      alt="check"
      src="check.png"
    />
    <h4>
      Thanks You! 
    </h4>
    <p>
      <b>
        Name :
      </b>
       
      
       
      
      <br />
      <b>
        Email :
      </b>
       
      
      <br />
      <b>
        Questions Attempted : 
      </b>
      <span
        className="number-score"
      >
        0
         /
         
        0
      </span>
      <br />
      <b>
        Score :
      </b>
       
      <span
        className="number-score"
      >
        0
         / 
        0
      </span>
    </p>
  </div>
  <p
    className="footer"
  >
    Copyright Â© 2020 , All Rights Reserved.
  </p>
</div>
`;
